src/dbnode/encoding/proto/encoder.go

```diff
@@ -308,24 +367,27 @@ func (enc *Encoder) reset(start time.Time, capacity int) {
 	enc.lastEncodedDP = ts.Datapoint{}
 	enc.unmarshaled = nil
 
+	// Prevent this from growing too large and remaining in the pools.
+	enc.marshalBuf = nil
+
 	if enc.schema != nil {
-		enc.customFields = customFields(enc.customFields, enc.schema)
+		enc.customFields, enc.protoFields = customFields(enc.customFields, enc.protoFields, enc.schema)
```

Comment:
should this be part of `Encoder`, since this exact state is being used in the same way in multiple places?